collectors/ciscoduo/collector.js

```diff
@@ -154,15 +154,24 @@ class CiscoduoCollector extends PawsCollector {
         else {
             // This condition works if next page getting null or undefined
             const untilMoment = moment();
-
-            const { nextUntilMoment, nextSinceMoment, nextPollInterval } = calcNextCollectionInterval('no-cap', untilMoment, this.pollInterval);
-
-            return {
-                stream: curState.stream,
-                mintime: nextSinceMoment.unix(),
-                poll_interval_sec: nextPollInterval
-            };
-
+            if (curState.mintime) {
+                // If next page is null check min time stamp is less than 1 hr move by 1 min else set the last hour time stamp as newMintime
+                const lastHourMoment = moment().subtract(1, 'hours').unix();
+                const newMintime = Math.max(curState.mintime + 1, lastHourMoment);
+                return {
+                    stream: curState.stream,
+                    mintime: newMintime,
+                    poll_interval_sec: 60
```

Comment:
I think if collector catches up with backlog and startc collecting last hour data then we can increase pool interval to 15 mins.